server/routes/goalsRouter.ts

```diff
@@ -0,0 +1,27 @@
+import { RequestHandler, Router } from 'express'
+import { Services } from '../services'
+import asyncMiddleware from '../middleware/asyncMiddleware'
+import getPrisonerData from '../middleware/getPrisonerDataMiddleware'
+import checkPrisonerInCaseload from '../middleware/checkPrisonerInCaseloadMiddleware'
+import GoalsController from '../controllers/goalsController'
+
+export default function goalsRouter(services: Services): Router {
+  const router = Router()
+
+  const get = (path: string | string[], ...handlers: RequestHandler[]) =>
+    router.get(
+      path,
+      handlers.map(handler => asyncMiddleware(handler)),
+    )
+
+  const goalsController = new GoalsController(services.workAndSkillsPageService)
+
+  get(
+    '/prisoner/:prisonerNumber/goals',
+    getPrisonerData(services, { minimal: true }),
+    checkPrisonerInCaseload(),
+    (req, res, next) => goalsController.displayGoals(req, res),
+  )
+
+  return router
+}
```

Comment:
Looking at how the main `work-and-skills` route is setup, it does a bit more than what we are doing in this route definition:
```
  get(
    '/prisoner/:prisonerNumber/work-and-skills',
    auditPageAccessAttempt({ services, page: Page.WorkAndSkills }),
    getPrisonerData(services),
    checkPrisonerInCaseload(),
    async (req, res, next) => {
      const prisonerData = req.middleware?.prisonerData
      const inmateDetail = req.middleware?.inmateDetail
      const { workAndSkillsPageService } = services
      const workAndSkillsPageData = await workAndSkillsPageService.get(res.locals.clientToken, prisonerData)

      const fullCourseHistoryLinkUrl = `${config.serviceUrls.digitalPrison}/prisoner/${prisonerData.prisonerNumber}/courses-qualifications`
      const workAndActivities12MonthLinkUrl = `${config.serviceUrls.digitalPrison}/prisoner/${prisonerData.prisonerNumber}/work-activities`
      const workAndActivities7DayLinkUrl = `${config.serviceUrls.digitalPrison}/prisoner/${prisonerData.prisonerNumber}/schedule`

      await services.auditService.sendPageView({
        userId: res.locals.user.username,
        userCaseLoads: res.locals.user.caseLoads,
        userRoles: res.locals.user.userRoles,
        prisonerNumber: prisonerData.prisonerNumber,
        prisonId: prisonerData.prisonId,
        correlationId: req.id,
        page: Page.WorkAndSkills,
      })

      /* New Work & Skills page introduced in epic RR-575
         Render the new view if the NEW_WORK_AND_SKILLS_TAB_ENABLED feature toggle is enabled
         TODO - The feature toggle and this ternary will be removed in RR-607
      */
      const viewTemplate = config.featureToggles.newWorkAndSkillsTabEnabled
        ? 'pages/new_workAndSkills'
        : 'pages/workAndSkills'

      res.render(viewTemplate, {
        ...mapHeaderData(prisonerData, inmateDetail, res.locals.user, 'work-and-skills'),
        ...workAndSkillsPageData,
        pageTitle: 'Work and skills',
        fullCourseHistoryLinkUrl,
        workAndActivities12MonthLinkUrl,
        workAndActivities7DayLinkUrl,
      })
    },
  )
```

Specifically I think it audits the page access attempt; and then (within the inline controller code) it send a page view event to the audit service.
IE. every time the main W&S page is displayed we would expect 1 or 2 audit records - 1 to audit the fact that I've attempted to view the page, and if the page view is successful another event to audit that I've actually viewed the page.

Do we need to do that here? I think we need to find out, but personally I'd suggest we do need to do it.
Have a look at `csraRouter` and `csraController` to see how they do it (ie. the CSRA pages are using the pattern of a separate router and controller class like we are doing here, but they also do this auditing)

Reply:
Added the auditing

Reply:
Awesome; thank you 👍